### ISSUE TITLE
[FEATURE]Support Kafka event processing

### ISSUE THREAD

**Issue by jmrozanec** [scikit-multiflow/scikit-multiflow#277]
**Is your feature request related to a problem? Please describe.**
We need to plug events streamed through Kafka to streaming algorithms so that upon arrival of Kafka messages, we can update train/predict using streaming algorithms. In particular, we are interested in using this library in a smart manufacturing context, to monitor and predict machine performance and downtimes.

**Describe the solution you'd like**
Provide some adaptor, that allows connecting to Kafka and stream the events to scikit-multiflow

**Describe alternatives you've considered**
We are developing a solution in [this fork](https://github.com/jmrozanec/scikit-multiflow), where we reframe scikit-multiflow to process events on demand instead of actively poll data sources for new events.

**Additional context**
We have industry requirements for streaming algorithms, but we experience an adoption barrier on the lack of support of scikit-multiflow to certain widely used stream sources, such as Kafka; and the fact that the library was designed to pull events, not to listen for them. Providing such adaptors would cover a wide range of new scenarios and ease the adoption of the library.

**Comment by MainRo** [comment]
Hello,
We have similar need, but follow another path: We use the model implementations from scikit-multiflow and integrate them directly in our framework that already supports kafka as a source.

The on-going work is available here and works with vanilla scikit-multiflow:
https://github.com/maki-nage/rxsci-multiflow

Feeding data from kafka is done with maki nage: makinage.org

It looks like several streaming/reactive frameworks want to integrate with scikit-multiflow (cf #106). It would be nice to make this possible without needing specific integration in scikit-multiflow. Using the different models directly is quite easy, but not the evaluators. I think that splitting the evaluator functions into smaller parts would make it: This would not change the existing scikit-multiflow evaluators API/behavior, and allow other frameworks to reuse most of this logic while only replacing the data feed part.

**Comment by jmrozanec** [comment]
@MainRo thank you for sharing! We agree regarding the evaluators behaviour. In our implementation we refactored them, providing different evaluation strategies.

**Comment by MainRo** [comment]
interesting, where did you move them ? The evaluation module is empty on your master.

**Comment by jmrozanec** [comment]
We have that in the skmultiflow/data/observer package. We thought about it as event observers with different trigger policies.

**Comment by garawalid** [comment]
+1 for this feature. This will be a great feature since Kafka is widely used in many companies. 

We can create a new issue to discuss the design of the new API of the evaluator in order to support other sources like Kafka, MQTT, or similar service.

I think it's better to work on the same project and add new features rather than create new forks.

**Comment by jmrozanec** [comment]
@garawalid we submitted a proposal in #278 which provides a working implementation. Seems the gh-actions are not detecting the package structure changed, thus failing the tests, which all run if executed locally.
The way we developed it, it is ready to support other sources such as RabbitMQ and others as well. I would suggest continuing the conversation on this issue, to avoid splitting the case over multiple issues.
